### ISSUE TITLE
Unable to configure Macbook Pro Trackpad

### ISSUE THREAD

**Issue by n8henrie** [SirCmpwn/sway#1059]
Hello,

I'm trying to configure my 2009 Macbook Pro trackpad to work with Sway under Arch Linux, using [this issue](https://github.com/SirCmpwn/sway/issues/365) as guidance.

With `swaymsg -t get_inputs`, I see:
```
 "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trac"
```

For comparison, with `sudo libinput-list-devices`, I see:

```
Device:           Apple Inc. Apple Internal Keyboard / Trackpad
```

In my sway config, I have: 
```
input "1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac" {
    tap enabled
    natural_scroll enabled
}
```

but it doesn't seem to be working -- scroll direction is still not "natural," tap-to-click doesn't work.

Thinking that perhaps the truncated name were a problem, I have tried all of the following with no luck (killing sway and restarting in between each):

```
 "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac" {
 "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trac" {
 "identifier": 1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac {
 "identifier": 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trac {
 "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad" {
 "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trackpad" {
 "identifier": 1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trackpad {
 "identifier": 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad {
```

What am I doing wrong?

**Comment by SirCmpwn** [comment]
Can you produce a debug log?

**Comment by n8henrie** [comment]
Wow, that was fast.

Is there a better way than `sway --verbose --debug &> sway_debug.log`?

Looks like the relevant portions:

```
01/25/17 13:53:03 - handling config command 'input 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad {'
01/25/17 13:53:03 - find_handler(input) 0
01/25/17 13:53:03 - new_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad)
01/25/17 13:53:03 - entering input block: 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad
01/25/17 13:53:03 - handling config command 'tap enabled'
01/25/17 13:53:03 - find_handler(tap) 1
01/25/17 13:53:03 - looking at input handlers
01/25/17 13:53:03 - tap for device: 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad
01/25/17 13:53:03 - new_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad)
01/25/17 13:53:03 - apply-tap for device: 1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad
01/25/17 13:53:03 - handling config command 'natural_scroll enabled'
01/25/17 13:53:03 - find_handler(natural_scroll) 1
01/25/17 13:53:03 - looking at input handlers
01/25/17 13:53:03 - new_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad)
01/25/17 13:53:03 - handling config command '}'
01/25/17 13:53:03 - End of input block
01/25/17 13:53:03 - handling config command 'bar {'
```

```
01/25/17 13:53:03 - Found input device (0:1:Power_Button)
01/25/17 13:53:03 - rewritten name Power_Button
01/25/17 13:53:03 - [wlc] Set new bsurface to output (1)
01/25/17 13:53:03 - [wlc] Added output (1)
01/25/17 13:53:03 - [wlc] become active
01/25/17 13:53:03 - rewritten name Power_Button
01/25/17 13:53:03 - Found input device (0:1:Power_Button)
01/25/17 13:53:03 - rewritten name Power_Button
01/25/17 13:53:03 - Found input device (0:1:Power_Button)
01/25/17 13:53:03 - rewritten name Sleep_Button
01/25/17 13:53:03 - Found input device (0:3:Sleep_Button)
01/25/17 13:53:03 - rewritten name Apple_Computer,_Inc._IR_Receiver
01/25/17 13:53:03 - Found input device (1452:33346:Apple_Computer,_Inc._IR_Re)
01/25/17 13:53:03 - rewritten name Apple_Inc._Apple_Internal_Keyboard_/_Trackpad
01/25/17 13:53:03 - Found input device (1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac)
01/25/17 13:53:03 - rewritten name bcm5974
01/25/17 13:53:03 - Found input device (1452:566:bcm)
01/25/17 13:53:03 - rewritten name Built-in_iSight
01/25/17 13:53:03 - new_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_\/_Trackpad)
01/25/17 13:53:03 - handling config command '}'
01/25/17 13:53:03 - End of input block
```

^^ As you can see, the above was with the config including the escape character `\`, and the full word `Trackpad`, but it behaves identically without the escape (I understand I am not supposed to include the `\`), and identically using `Trac` instead of `Trackpad` (as per `swaymsg` output).

**Comment by SirCmpwn** [comment]
Use ` "identifier": "1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac" {` and provide a full debug log, please.

**Comment by n8henrie** [comment]
Sounds good. Sorry, having a hard time figuring out how to pipe output to the clipboard (in order to paste to browser). Coming from OSX would be `cat myfile.log | pbcopy`. GitHub won't let me upload the file either (`We don’t support that file type.` with both `.log` and `.txt` extension.)

Also, as per above is `sway --verbose --debug &> out.log` basically the format you were expecting?
```
01/25/17 18:12:14 - [wlc] logind: session control granted
01/25/17 18:12:14 - [wlc] Running on vt 1 (fd 0)
01/25/17 18:12:14 - [wlc] Failed to open X11 display
01/25/17 18:12:14 - Starting sway version  (2016-10-25, branch "")

01/25/17 18:12:14 - Set XDG_CONFIG_HOME to /home/n8henrie/.config
01/25/17 18:12:14 - Loading config from /home/n8henrie/.config/sway/config
01/25/17 18:12:14 - handling config command 'set $mod Mod4'
01/25/17 18:12:14 - find_handler(set) 0
01/25/17 18:12:14 - handling config command 'set $term urxvt'
01/25/17 18:12:14 - find_handler(set) 0
01/25/17 18:12:14 - handling config command 'set $menu dmenu_run'
01/25/17 18:12:14 - find_handler(set) 0
01/25/17 18:12:14 - handling config command 'output * bg /usr/share/sway/Sway_Wallpaper_Blue_1920x1080.png fill'
01/25/17 18:12:14 - find_handler(output) 0
01/25/17 18:12:14 - Config stored for output * (enabled:-1) (-1 x -1 @ -1, -1 scale 1) (bg /usr/share/sway/Sway_Wallpaper_Blue_1920x1080.png fill)
01/25/17 18:12:14 - handling config command 'bindsym $mod+Return exec $term'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Return to command exec urxvt
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+q kill'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+q to command kill
01/25/17 18:12:14 - handling config command 'bindsym $mod+d exec $menu'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+d to command exec dmenu_run
01/25/17 18:12:14 - handling config command 'floating_modifier $mod normal'
01/25/17 18:12:14 - find_handler(floating_modifier) 0
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+c reload'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+c to command reload
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+e exit'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+e to command exit
01/25/17 18:12:14 - handling config command 'bindsym $mod+h focus left'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+h to command focus left
01/25/17 18:12:14 - handling config command 'bindsym $mod+j focus down'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+j to command focus down
01/25/17 18:12:14 - handling config command 'bindsym $mod+k focus up'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+k to command focus up
01/25/17 18:12:14 - handling config command 'bindsym $mod+l focus right'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+l to command focus right
01/25/17 18:12:14 - handling config command 'bindsym $mod+Left focus left'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Left to command focus left
01/25/17 18:12:14 - handling config command 'bindsym $mod+Down focus down'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Down to command focus down
01/25/17 18:12:14 - handling config command 'bindsym $mod+Up focus up'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Up to command focus up
01/25/17 18:12:14 - handling config command 'bindsym $mod+Right focus right'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Right to command focus right
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+h move left'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+h to command move left
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+j move down'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+j to command move down
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+k move up'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+k to command move up
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+l move right'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+l to command move right
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+Left move left'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+Left to command move left
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+Down move down'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+Down to command move down
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+Up move up'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+Up to command move up
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+Right move right'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+Right to command move right
01/25/17 18:12:14 - handling config command 'bindsym $mod+1 workspace 1'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+1 to command workspace 1
01/25/17 18:12:14 - handling config command 'bindsym $mod+2 workspace 2'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+2 to command workspace 2
01/25/17 18:12:14 - handling config command 'bindsym $mod+3 workspace 3'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+3 to command workspace 3
01/25/17 18:12:14 - handling config command 'bindsym $mod+4 workspace 4'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+4 to command workspace 4
01/25/17 18:12:14 - handling config command 'bindsym $mod+5 workspace 5'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+5 to command workspace 5
01/25/17 18:12:14 - handling config command 'bindsym $mod+6 workspace 6'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+6 to command workspace 6
01/25/17 18:12:14 - handling config command 'bindsym $mod+7 workspace 7'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+7 to command workspace 7
01/25/17 18:12:14 - handling config command 'bindsym $mod+8 workspace 8'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+8 to command workspace 8
01/25/17 18:12:14 - handling config command 'bindsym $mod+9 workspace 9'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+9 to command workspace 9
01/25/17 18:12:14 - handling config command 'bindsym $mod+0 workspace 10'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+0 to command workspace 10
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+1 move container to workspace 1'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+1 to command move container to workspace 1
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+2 move container to workspace 2'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+2 to command move container to workspace 2
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+3 move container to workspace 3'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+3 to command move container to workspace 3
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+4 move container to workspace 4'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+4 to command move container to workspace 4
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+5 move container to workspace 5'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+5 to command move container to workspace 5
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+6 move container to workspace 6'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+6 to command move container to workspace 6
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+7 move container to workspace 7'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+7 to command move container to workspace 7
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+8 move container to workspace 8'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+8 to command move container to workspace 8
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+9 move container to workspace 9'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+9 to command move container to workspace 9
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+0 move container to workspace 10'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+0 to command move container to workspace 10
01/25/17 18:12:14 - handling config command 'bindsym $mod+b splith'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+b to command splith
01/25/17 18:12:14 - handling config command 'bindsym $mod+v splitv'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+v to command splitv
01/25/17 18:12:14 - handling config command 'bindsym $mod+s layout stacking'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+s to command layout stacking
01/25/17 18:12:14 - handling config command 'bindsym $mod+w layout tabbed'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+w to command layout tabbed
01/25/17 18:12:14 - handling config command 'bindsym $mod+e layout toggle split'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+e to command layout toggle split
01/25/17 18:12:14 - handling config command 'bindsym $mod+f fullscreen'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+f to command fullscreen
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+space floating toggle'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+Shift+space to command floating toggle
01/25/17 18:12:14 - handling config command 'bindsym $mod+space focus mode_toggle'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+space to command focus mode_toggle
01/25/17 18:12:14 - handling config command 'bindsym $mod+a focus parent'
01/25/17 18:12:14 - find_handler(bindsym) 0
01/25/17 18:12:14 - bindsym - Bound Mod4+a to command focus parent
01/25/17 18:12:14 - handling config command 'bindsym $mod+Shift+minus move scratchpad'
01/25/17 18:12:14 - find_handler(bindsym) 0
...skipping...
: 800|x:   0|y:   0|g:  0|vis:t|children: 1|name:root
|focus:X     |(0xba36c0)(p:0x63a4c0)(f:0xbc56b0)(h: 1)Type:op  |layout:-    |w:1280|h: 800|x:   0|y:   0|g: -1|vis:t|children: 1|name:LVDS-1
|-focus:W    |(0xbc56b0)(p:0xba36c0)(f:0x1052ad0)(h:18446744073709551615)Type:ws  |layout:Horiz|w:1280|h: 781|x:   0|y:  19|g: -1|vis:t|children: 2|name:1
|--focus:X   |(0x103cd30)(p:0xbc56b0)(f:(nil)   )(h: 5)Type:view|layout:-    |w: 640|h: 781|x:   0|y:  19|g: -1|vis:t|children: 0|name:Unable to config
|--focus:X   |(0x102e980)(p:0xbc56b0)(f:(nil)   )(h: 4)Type:view|layout:-    |w: 640|h: 781|x: 640|y:  19|g: -1|vis:t|children: 0|name:urxvt
|==focus:K   |(0x1052ad0)(p:0xbc56b0)(f:(nil)   )(h: 6)Type:view|layout:-    |w:   1|h:   1|x: 640|y: 390|g: -1|vis:t|children: 0|name:Open Files
01/25/17 18:17:08 - geometry request for 6 236x214 @ 639,390
01/25/17 18:17:08 - Sane values for view to 236 x 214 @ 640, 390
01/25/17 18:17:09 - Setting focus to 0x1052ad0:6 (VIEW 'Open Files')
01/25/17 18:17:09 - Sending window::focus event
01/25/17 18:17:10 - Setting focus to 0x102e980:4 (VIEW 'urxvt')
01/25/17 18:17:10 - Sending window::focus event
01/25/17 18:17:11 - Setting focus to 0x1052ad0:6 (VIEW 'Open Files')
01/25/17 18:17:11 - Sending window::focus event
01/25/17 18:17:11 - Destroying window 6
01/25/17 18:17:11 - Destroying view '0x1052ad0'
01/25/17 18:17:11 - Arranging layout for 0xbc56b0 (WORKSPACE '1')
01/25/17 18:17:11 - Arranging layout for 0xbc56b0 1 1280.000000x781.000000+0.000000,19.000000
01/25/17 18:17:11 - -> Found panel for this workspace: 1280x19, position: 0
01/25/17 18:17:11 - Arranging workspace '1' at 0.000000, 19.000000
01/25/17 18:17:11 - Arranging 0xbc56b0 horizontally
01/25/17 18:17:11 - Calculating arrangement for 0x103cd30:4 (will scale 1280.000000 by 1.000000)
01/25/17 18:17:11 - Arranging layout for 0x103cd30 Unable to configure Macbook Pro Trackpad · Issue #1059 · SirCmpwn/sway - Chromium 640.000000x781.000000+0.000000,19.000000
01/25/17 18:17:11 - Set view to 640 x 781 @ 0, 19
01/25/17 18:17:11 - Calculating arrangement for 0x102e980:4 (will scale 1280.000000 by 1.000000)
01/25/17 18:17:11 - Arranging layout for 0x102e980 urxvt 640.000000x781.000000+640.000000,19.000000
01/25/17 18:17:11 - Set view to 640 x 781 @ 640, 19
focus:R      |(0x63a4c0)(p:(nil)   )(f:0xba36c0)(h:18446744073709551615)Type:root|layout:-    |w:1280|h: 800|x:   0|y:   0|g:  0|vis:t|children: 1|name:root
|focus:X     |(0xba36c0)(p:0x63a4c0)(f:0xbc56b0)(h: 1)Type:op  |layout:-    |w:1280|h: 800|x:   0|y:   0|g: -1|vis:t|children: 1|name:LVDS-1
|-focus:W    |(0xbc56b0)(p:0xba36c0)(f:0x103cd30)(h:18446744073709551615)Type:ws  |layout:Horiz|w:1280|h: 781|x:   0|y:  19|g: -1|vis:t|children: 2|name:1
|--focus:K   |(0x103cd30)(p:0xbc56b0)(f:(nil)   )(h: 5)Type:view|layout:-    |w: 640|h: 781|x:   0|y:  19|g: -1|vis:t|children: 0|name:Unable to config
|--focus:X   |(0x102e980)(p:0xbc56b0)(f:(nil)   )(h: 4)Type:view|layout:-    |w: 640|h: 781|x: 640|y:  19|g: -1|vis:t|children: 0|name:urxvt
01/25/17 18:17:11 - Sending window::close event
01/25/17 18:17:11 - Setting focus to 0x103cd30:5 (VIEW 'Unable to configure Macbo)
01/25/17 18:17:11 - Sending window::focus event
01/25/17 18:17:14 - Setting focus to 0x102e980:4 (VIEW 'urxvt')
01/25/17 18:17:14 - Sending window::focus event
01/25/17 18:17:15 - Setting focus to 0x103cd30:5 (VIEW 'Unable to configure Macbo)
01/25/17 18:17:15 - Sending window::focus event
01/25/17 18:18:14 - Setting focus to 0x102e980:4 (VIEW 'urxvt')
01/25/17 18:18:14 - Sending window::focus event
01/25/17 18:18:15 - Setting focus to 0x103cd30:5 (VIEW 'Unable to configure Macbo)
01/25/17 18:18:15 - Sending window::focus event
01/25/17 18:18:15 - Setting focus to 0x102e980:4 (VIEW 'urxvt')
01/25/17 18:18:15 - Sending window::focus event
```

**Comment by SirCmpwn** [comment]
Quit doctoring it and just upload the entire log to a pastebin somewhere like gist.github.com. And your sway config. Also, you're a version behind.

**Comment by n8henrie** [comment]
Sorry, I didn't do any "doctoring" intentionally -- opened the log file in one window in `less`, highlighted, copied, and pasted directly into the comment box.

https://gist.github.com/n8henrie/10e3232cc31373ce75cd046bb8ebab8d

Again, I'll assume `sway --verbose --debug &> out.log` is a reasonable way to get the information you're asking for unless you tell me otherwise (I don't see any mention of another log file, a flag or envvar for another  debug output location). Thanks for your time.

**Comment by SirCmpwn** [comment]
"...skipping..."

Said doctoring isn't present in the new gist, which is indeed a complete log, so I can identify the issue. The important part appears to be here:

```
01/25/17 21:53:33 - [handlers.c:131] Found input device (1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac)
01/25/17 21:53:33 - [handlers.c:141] Matched input config for 1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac
01/25/17 21:53:33 - [config.c:969] apply_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac)
01/25/17 21:53:33 - [config.c:996] apply_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac) natural_scroll_set_enabled(1)
01/25/17 21:53:33 - [config.c:1012] apply_input_config(1452:566:Apple_Inc._Apple_Internal_Keyboard_/_Trac) tap_set_enabled(1)
```

It seems that the device is correctly identified, your config is matched, and your options applied. This indicates that the problem is in libinput, not in sway - your options are being faithfully passed to libinput by sway. Report to them for assistance. Good luck!

**Comment by n8henrie** [comment]
Figured it out, it was a sway config issue (my fault). For future Googlers:

Looking at the capabilities listed under `sudo libinput-list-devices`, it showed that the device I was trying to use above doesn't have `Tap-to-click` or `Nat.scrolling`, which surprised me. Looking below that, it showed a device named `bcm5974` that *did* have those capabilities. I googled and ends up that's the touchpad driver. I changed my config to use `input "1452:566:bcm"` (as reported by `swaymsg -t get_inputs`) and it worked instantly.

Also, I've reproduced the `...skipping...` issue -- it's from scrolling while highlighting in `less`. Sorry about that.
